activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskAdminRuntimeImpl.java

```diff
@@ -132,4 +138,136 @@ public Task release(ReleaseTaskPayload releaseTaskPayload) {
+                                                u);
+            }
+        }
+    }
+    
+    @Override
+    public void addCandidateGroups(CandidateGroupsPayload candidateGroupsPayload) {          
+        Task task;
+        try {
+            task = task(candidateGroupsPayload.getTaskId()); 
+            
+        } catch (IllegalStateException ex) {
+            throw new IllegalStateException("The authenticated user cannot update the task" + candidateGroupsPayload.getTaskId() + " due it is not the current assignee");
+        }
+              
+        if (candidateGroupsPayload.getCandidateGroups() != null && !candidateGroupsPayload.getCandidateGroups().isEmpty()) {
+            for (String g : candidateGroupsPayload.getCandidateGroups()) {
+                taskService.addCandidateGroup(task.getId(),
+                                              g);
+            }
+        }
+    }
+    
+    @Override
+    public void deleteCandidateGroups(CandidateGroupsPayload candidateGroupsPayload) {    
+        Task task;
+        try {
+            task = task(candidateGroupsPayload.getTaskId()); 
+            
+        } catch (IllegalStateException ex) {
+            throw new IllegalStateException("The authenticated user cannot update the task" + candidateGroupsPayload.getTaskId() + " due it is not the current assignee");
+        }
+
+       
+        if (candidateGroupsPayload.getCandidateGroups() != null && !candidateGroupsPayload.getCandidateGroups().isEmpty()) {
+            for (String g : candidateGroupsPayload.getCandidateGroups()) {
+                taskService.deleteCandidateGroup(task.getId(),
+                                                 g);
+            }
+        }
+    }
+    
+    @Override
+    public List<String> userCandidates(String taskId) {
+        List<IdentityLink> identityLinks= getIdentityLinks(taskId);
+        List<String> userCandidates = new ArrayList<String>();
+        if (identityLinks!=null) {
+            for (IdentityLink i : identityLinks) {
+                if (i.getUserId()!=null) {
+                    if (i.getType().equals(IdentityLinkType.CANDIDATE) ) {
+                        userCandidates.add(i.getUserId());
+                    } 
+                }
+            }
+            
+        }
+        return userCandidates;
+    }
+    
+    @Override
+    public List<String> groupCandidates(String taskId) {
+        List<IdentityLink> identityLinks= getIdentityLinks(taskId);
+        List<String> groupCandidates = new ArrayList<String>();
+        if (identityLinks!=null) {
+            for (IdentityLink i : identityLinks) {
+                if (i.getGroupId()!=null) {
+                    if (i.getType().equals(IdentityLinkType.CANDIDATE) ) {
+                        groupCandidates.add(i.getGroupId());
+                    } 
+                }
+            }
+
+        }
+        return groupCandidates;
+    }
+    
+    private List<IdentityLink> getIdentityLinks(String taskId) {
```

Comment:
@CTI777 this is ok.. because it is private.. we shouldn't expose IdentityLinks in our APIs, so no action needed.. just for you to know